### ISSUE TITLE
NUnit2025 fires unnecessarily

### ISSUE THREAD

**Issue by manfred-brands** [nunit/nunit.analyzers#621]
Given:
```csharp
var item1 = new object();
var item2 = "Hello";
var collection = new[] { item1, item2 };

Assert.That(collection, Does.Contain(item1));
Assert.That(collection, Does.Contain(item2));
```

NUnit2025 fires for the second line: _The contains constraint cannot be used with an actual value of 'object[]'_.
However the tests succeed in NUnit.

In the analyzer, the rule fires always if _actual_ is neither a `string` nor an `IEnumerable<string>`.

The `ContainsClassicModelAssertUsageCodeFix` converts `Assert.Contain` into `Does.Contain`.
I made the latter because that was the code we used a lot in our projects.
As NUnit has multiple ways to do the same thing, maybe the  it is the wrong method, but NUnit itself considers it ok, 
hence so should the analyzer.

Note that `Contains.Item` has no equivalent for *not*, e.g. `Contains.No.Item` does not exist.
We do have `Has.Member` and `Has.No.Member`.

**Comment by manfred-brands** [comment]
@mikkelbu Do you agree to fix the analyzer or should `Does.Contain` not be used for other collections?

**Comment by manfred-brands** [comment]
There was an explicit commit to allow Does.Contain` for collections:
```
Author: Rob Prouse <rob@prouse.org>
Date: 26/08/2014 04:42:06
Commit hash: 88cbece5c5c40d4dccf368453cbc0f0d9fb043f2

Added collections to the Does.Contain syntax and cleaned up the various Contain/Contains methods to return the right type of Contraints for strings and collections.
```

**Comment by mikkelbu** [comment]
@manfred-brands From what you have written above it sounds like the analyzer should be disabled
